From a582823bba7a7610a1bbb9bd467c8311f1cf087b Mon Sep 17 00:00:00 2001
From: Esme Povirk <esme@codeweavers.com>
Date: Sat, 6 Dec 2025 19:14:25 +0000
Subject: [PATCH 001/410] mscoree: HACK: Accept outdated support msi outside
 prefix.

This prevents Wine from downloading and installing the runtime msi
when the Wine tree has a newer Wine Mono version than the Proton
tree.
---
 dlls/mscoree/mscoree_main.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/dlls/mscoree/mscoree_main.c b/dlls/mscoree/mscoree_main.c
index 2b9f78a3955..b68f9a50969 100644
--- a/dlls/mscoree/mscoree_main.c
+++ b/dlls/mscoree/mscoree_main.c
@@ -800,7 +800,7 @@ static BOOL invoke_appwiz(void)
     return ret;
 }
 
-static BOOL get_support_msi(LPCWSTR mono_path, LPWSTR msi_path)
+static BOOL get_support_msi(LPCWSTR mono_path, LPWSTR msi_path, BOOL ignore_version)
 {
     static const WCHAR support_msi_relative[] = {'\\','s','u','p','p','o','r','t','\\','w','i','n','e','m','o','n','o','-','s','u','p','p','o','r','t','.','m','s','i',0};
     UINT (WINAPI *pMsiOpenPackageW)(LPCWSTR,ULONG*);
@@ -838,7 +838,7 @@ static BOOL get_support_msi(LPCWSTR mono_path, LPWSTR msi_path)
     if (res == ERROR_SUCCESS) {
         TRACE("found support msi version %s at %s\n", versionstringbuf, debugstr_w(msi_path));
 
-        if (compare_versions(WINE_MONO_VERSION, versionstringbuf) <= 0)
+        if (ignore_version || compare_versions(WINE_MONO_VERSION, versionstringbuf) <= 0)
         {
             ret = TRUE;
         }
@@ -922,12 +922,12 @@ static BOOL install_wine_mono(void)
 
     initresult = CoInitialize(NULL);
 
-    ret = get_support_msi(mono_path, support_msi_path);
+    ret = get_support_msi(mono_path, support_msi_path, FALSE);
     if (!ret)
     {
         /* Try looking outside c:\windows\mono */
         ret = (get_mono_path(mono_path, TRUE) &&
-            get_support_msi(mono_path, support_msi_path));
+            get_support_msi(mono_path, support_msi_path, TRUE));
     }
 
     if (ret)
-- 
2.51.1

